### PR TITLE
New doc for Event analyzer

### PR DESCRIPTION
**Pull request by nvanthao**
New doc for https://github.com/replicatedhq/troubleshoot/pull/1474

### CODE REVIEW

File: docs/source/analyze/event.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+The `Event` analyzer checks if an Event is existed within the cluster resources in a given namespace.
+
+The analyzer uses data from the [clusterResources collector](https://troubleshoot.sh/collect/cluster-resources).
+The `clusterResources` collector is automatically added and will always be present.
+
+The target Event can be identified by `Reason`, `Kind` or a regular expression matching the Event `Message`.
+
+The outcomes on this analyzer will be processed in order, and execution will stop after the first outcome that is truthy.
+
+The analyzer also has access to all fields in the Event [object](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#event-v1-core), Go templating can be used for dynamic message. E.g. `{{ .Reason }} {{ .InvovledObject.Name }}`
```
Comment (banjoh) [inline docs/source/analyze/event.md:15]:
```suggestion
The analyzer also has access to all fields in the Event [object](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/), Go templating can be used for dynamic message. E.g. `{{ .Reason }} {{ .InvovledObject.Name }}`
```

**Review by banjoh — CHANGES_REQUESTED** [review]

File: docs/source/analyze/event.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+title: Event
+description: Perform analysis and checks on Kubernetes Events found in the cluster
+---
+
+The `Event` analyzer checks if an Event is existed within the cluster resources in a given namespace.
```
Comment (paigecalvert) [inline docs/source/analyze/event.md:6]:
```suggestion
The `Event` analyzer checks if an Event exists within the cluster resources in a given namespace.
```

File: docs/source/analyze/event.md (hunk 1 above)
Comment (paigecalvert) [inline docs/source/analyze/event.md:15]:
```suggestion
The analyzer also has access to all fields in the Event [object](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#event-v1-core). Go templating can be used for dynamic message. For example, `{{ .Reason }} {{ .InvovledObject.Name }}`
```

File: docs/source/analyze/event.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The target Event can be identified by `Reason`, `Kind` or a regular expression matching the Event `Message`.
+
+The outcomes on this analyzer will be processed in order, and execution will stop after the first outcome that is truthy.
+
+The analyzer also has access to all fields in the Event [object](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#event-v1-core), Go templating can be used for dynamic message. E.g. `{{ .Reason }} {{ .InvovledObject.Name }}`
+
+## Parameters
+
+**reason**: (Required) Event Reason. E.g. `InvalidDiskCapacity`. Possible reasons list can be referred from Kubernetes [source code](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/events/event.go)
```
Comment (paigecalvert) [inline docs/source/analyze/event.md:19]:
```suggestion
**reason**: (Required) Event Reason. For example, `InvalidDiskCapacity`. Possible reasons list can be referred from Kubernetes [source code](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/events/event.go)
```

File: docs/source/analyze/event.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+**namespace**: (Optional) The namespace to look for the deployment in.
+If specified, analysis will be limited to deployments in this namespace.
+
+**kind**: (Optional) The REST resource the Event represents. E.g. `Pod`
```
Comment (paigecalvert) [inline docs/source/analyze/event.md:24]:
```suggestion
**kind**: (Optional) The REST resource the Event represents. For example,`Pod`
```

**Review by paigecalvert — APPROVED** [review]
looks good! just noticed a couple typos and I also suggested changing instances of `E.g.` to `For example`

File: docs/source/analyze/event.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+title: Event
+description: Perform analysis and checks on Kubernetes Events found in the cluster
+---
+
+The `Event` analyzer checks if an Event is existed within the cluster resources in a given namespace.
+
+The analyzer uses data from the [clusterResources collector](https://troubleshoot.sh/collect/cluster-resources).
+The `clusterResources` collector is automatically added and will always be present.
```
[the remark this answers is not in the document]
Comment (nvanthao) [reply docs/source/analyze/event.md:9]: I copied the text from `deploymentStatus` and believe the analysis currently looks the same for others analyzers such as `statefulsetStatus`, `replicasetStatus`, `jobStatus` etc. I reckon we can discuss this in other PR to include `ignoreIfNoFiles` to more analyzers. Please let me know your thought.

**Review by banjoh — APPROVED** [review]

File: docs/source/analyze/event.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (banjoh) [reply docs/source/analyze/event.md:9]: Thinking about it some more, I think its fine to leave it as is. Anyone defining this analyser expects cluster resources to exist

**Pull request merged**